magic/compat.py

```diff
@@ -16,6 +17,13 @@
 
 _libraries = {}
 _libraries['magic'] = loader.load_lib()
+magic_file = None
+if not os.environ.get("MAGIC"):
+    # wheels package the mime database in this directory
+    # prefer it when no magic file is specified by the user
+    mime_db = os.path.join(os.path.dirname(__file__), 'magic.mgc')
+    if os.path.exists(mime_db):
+        magic_file = mime_db
```

Comment:
one way to avoid code duplication (and cyclic import) would be to add `from magic.compat import magic_file` on top of `__init__.py`, and use it as default argument: 
```py
Magic.__init__(magic_file=magic_file, ...)
```

option 2 would be to put the snippet in a new file and import it in both places (avoids cyclic import as well).

any preference here @ahupp?

Reply:
> add `from magic.compat import magic_file` on top of `__init__.py`,

I don't think that's a good idea, as it means that it'd be impossible to import any part of python-magic without also importing `magic.compat`. A `compat` module, specifically, should only be imported when needed.

The code _could_ be moved to the top level of `__init__.py`, and imported into `compat.py` with `from . import magic_file`. I think that would be a better direction for things to flow, if going that route.

Reply:
Oh, never mind. `__init__.py` already unconditionally imports `magic.compat` in its `_add_compat()` function. Which is always run on import.